File: iotcloud-restapi-php/Core/src/IoTCloud/Response/CreateDeviceResponse.php
<?php

namespace TXIoTCloud\Response;

require_once "BaseResponse.php";

class CreateDeviceResponse extends BaseResponse
{
    /**
     * 设备名称
     */
    private $deviceName;

    /**
     * 设备证书，用于 TLS 建立链接时校验客户端身份
     */
    private $deviceCert;

    /**
     * 设备私钥，用于 TLS 建立链接时校验客户端身份，腾讯云后台不保存，请妥善保管
     */
    private $devicePrivateKey;

    public function __construct()
    {
    }

    /**
     * 获取设备名称
     */
    public function getDeviceName()
    {
        return $this->deviceName;
    }

    /**
     * 设置设备名称
     * @param
     */
    public function setDeviceName($deviceName)
    {
        $this->deviceName = $deviceName;
    }

    /**
     * 获取设备证书
     */
    public function getDeviceCert()
    {
        return $this->deviceCert;
    }

    /**
     * 设置设备证书
     * @param $deviceCert
     */
    public function setDeviceCert($deviceCert)
    {
        $this->deviceCert = $deviceCert;
    }

    /**
     * 获取设备私钥
     */
    public function getDevicePrivateKey()
    {
        return $this->devicePrivateKey;
    }

    /**
     * 设置设备私钥
     * @param $devicePrivateKey
     */
    public function setDevicePrivateKey($devicePrivateKey)
    {
        $this->devicePrivateKey = $devicePrivateKey;
    }

}

File: iotcloud-restapi-java/Core/src/com/qcloud/restapi/iotcore/response/CreateDeviceResponse.java
package com.qcloud.restapi.iotcore.response;

import com.google.gson.*;

public class CreateDeviceResponse extends Response {

    /**
     * 解析JSON字串并构造CreateDeviceResponse对象
     *
     * @param response 从服务端返回的应答JSON字串
     */
    public CreateDeviceResponse(String response) {
        super(response);
    }

    public String getDeviceName() {
        try {
            if (jsonObj != null) {
                return jsonObj.get("deviceName").getAsString();
            }
        }catch (Exception e) {
        }
        return "";
    }

    public String getDeviceCert() {
        try {
            if (jsonObj != null) {
                return jsonObj.get("deviceCert").getAsString();
            }
        }catch (Exception e) {
        }

        return "";
    }

    public String getDevicePrivateKey() {
        try {
            if (jsonObj != null) {
                return jsonObj.get("devicePrivateKey").getAsString();
            }
        }catch (Exception e) {
        }

        return "";
    }
}
